frontend/src/lib/components/elements/Toast/Toast.test.tsx

```diff
@@ -0,0 +1,174 @@
+    placement={PLACEMENT.bottomRight}
+    // increasing autoHideDuration to 10s to avoid test flakiness
+    autoHideDuration={10000}
+    overrides={{
+      Root: {
+        props: {
+          "data-testid": "toastContainer",
+        },
+      },
+    }}
+  />
+)
+
+const getProps = (
+  elementProps: Partial<ToastProto> = {},
+  themeProps: Partial<EmotionTheme> = {}
+): ToastProps => ({
+  text: "This is a toast message",
+  icon: "🐶",
+  theme: {
+    ...mockTheme.emotion,
+    ...themeProps,
+  },
+  width: 0,
+  ...elementProps,
+})
+
+const renderComponent = (props: ToastProps): RenderResult => {
+  return render(
+    <>
+      {createContainer()}
+      <Toast {...props} />
+    </>
+  )
+}
+
+describe("Toast Component", () => {
+  test("renders default toast", () => {
+    const props = getProps()
+    renderComponent(props)
+
+    const toast = screen.getByRole("alert")
+    const closeButton = screen.getByRole("button", { name: "Close" })
+    const expandButton = screen.queryByRole("button", { name: "view more" })
+
+    expect(toast).toBeInTheDocument()
+    expect(toast).toHaveTextContent("🐶 This is a toast message")
+    expect(closeButton).toBeInTheDocument()
+    expect(expandButton).not.toBeInTheDocument()
+  })
+
+  test("renders long toast messages with expand option", () => {
+    const props = getProps({
+      icon: "",
+      text: "Random toast message that is a really really really really really really really really really long message, going way past the 3 line limit",
+    })
+    renderComponent(props)
+
+    const toast = screen.getByRole("alert")
+    const expandButton = screen.getByRole("button", { name: "view more" })
+    expect(toast).toBeInTheDocument()
+    expect(toast).toHaveTextContent(
+      "Random toast message that is a really really really really really really really really really long message,"
+    )
+    expect(toast).toContainElement(expandButton)
+  })
+
+  test("can expand to see full toast message", () => {
+    const props = getProps({
+      icon: "",
+      text: "Random toast message that is a really really really really really really really really really long message, going way past the 3 line limit",
+    })
+    renderComponent(props)
+
+    const toast = screen.getByRole("alert")
+    const expandButton = screen.getByRole("button", { name: "view more" })
+    expect(toast).toBeInTheDocument()
+    expect(toast).toHaveTextContent(
+      "Random toast message that is a really really really really really really really really really long message,"
+    )
+    expect(toast).toContainElement(expandButton)
+
+    fireEvent.click(expandButton)
+    expect(toast).toHaveTextContent(
+      "Random toast message that is a really really really really really really really really really long message, going way past the 3 line limit"
+    )
+    const collapseButton = screen.getByRole("button", { name: "view less" })
+    expect(toast).toContainElement(collapseButton)
+  })
```

Comment:
nit: I think it'd be fine to combine the expand + collapse tests since they're so similar + the expand test is nearly part of the collapse test already

Reply:
Consolidated!